### PR TITLE
Fix: Keep packages sorted

### PR DESCRIPTION
**Pull request by localheinz**
### Description

By configuring the `sort-packages` option in `composer.json`

* requiring _or_
* removing

a dependency will automatically trigger sorting of the packages in the corresponding

* `require`
* `require-dev`

sections.

Of course, as an alternative to configuring this in `composer.json`, the `--sort-packages` flag can be used manually.

Keeping these packages sorted has the following advantages:

* it's easier to scan a list for items when it is sorted
* due to the nature of sorting, the more important platform requirements will rise to the top

Since there is no command for adding or removing items to or from the `replace` section, I sorted those manually using the [Lines Sorter](https://plugins.jetbrains.com/plugin/5919-lines-sorter) plugin for PhpStorm.

💁‍♂️ For reference, see 

- https://getcomposer.org/doc/06-config.md#sort-packages
- https://getcomposer.org/doc/03-cli.md#require

### Fixed Issues (if relevant)

n/a

### Manual testing scenarios

n/a
### Contribution checklist

 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds on Travis CI are green)This PR

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=10769) <br/>All committers have signed the CLA.

File: composer.json [hunk 1]
```diff
@@ -7,80 +7,83 @@
         "OSL-3.0",
         "AFL-3.0"
     ],
+    "config": {
+        "sort-packages": true
```
Comment (localheinz) [inline composer.json]: Here's what I ran after adding this section to trigger sorting packages (without updating any dependencies):

```
$ composer require composer/composer:1.4.1
$ composer require --dev squizlabs/php_codesniffer:3.0.1
```

File: composer.json [hunk 2]
```diff
@@ -7,83 +7,96 @@
...
         "lusitanian/oauth": "~0.8.10",
-        "sebastian/phpcpd": "2.0.4"
+        "pdepend/pdepend": "2.5.0",
+        "phpmd/phpmd": "@stable",
+        "phpunit/phpunit": "~6.2.0",
+        "sebastian/phpcpd": "2.0.4",
+        "squizlabs/php_codesniffer": "3.0.1"
     },
     "replace": {
-        "magento/module-marketplace": "100.3.0-dev",
```
Comment (localheinz) [inline composer.json]: Here's what I did after sorting packages in this section using https://plugins.jetbrains.com/plugin/5919-lines-sorter:

* ran `composer validate`
* fixed the trailing comma issue
* ran `composer update --lock`

For reference, see:

- https://getcomposer.org/doc/03-cli.md#validate
- https://getcomposer.org/doc/03-cli.md#update

**Comment by localheinz** [comment]
There's quite a bunch of `composer.json` checked in, should the change be applied to all of them?

**Comment by dmanners** [comment]
@localheinz if you feel up for it then you could update all the composer.json files that are checked in.

**Comment by dmanners** [comment]
Thanks for the update @localheinz could you take a moment to look into the conflict on the file `app/code/Magento/Backend/composer.json` for me. Thank you.

**Comment by localheinz** [comment]
@dmanners 

Done!

File: dev/tests/acceptance/tests/functional/Magento/FunctionalTest/Analytics/composer.json [hunk 3]
```diff
@@ -7,14 +7,17 @@
...
     "require": {
         "php": "~7.0",
-        "codeception/codeception": "2.2|2.3",
         "allure-framework/allure-codeception": "dev-master",
+        "codeception/codeception": "2.2|2.3",
         "consolidation/robo": "^1.0.0",
         "henrikbjorn/lurker": "^1.2",
-        "vlucas/phpdotenv": "~2.4",
         "magento/magento2-functional-testing-framework": "dev-develop"
+        "vlucas/phpdotenv": "~2.4",
```
Comment (dmanners) [inline dev/tests/acceptance/tests/functional/Magento/FunctionalTest/Analytics/composer.json]: Seems to be a miss `,` the line above.

File: dev/tests/acceptance/tests/functional/Magento/FunctionalTest/Robots/composer.json [hunk 4]
```diff
@@ -7,14 +7,17 @@
...
     "require": {
         "php": "~7.0",
-        "codeception/codeception": "2.2|2.3",
         "allure-framework/allure-codeception": "dev-master",
+        "codeception/codeception": "2.2|2.3",
         "consolidation/robo": "^1.0.0",
         "henrikbjorn/lurker": "^1.2",
-        "vlucas/phpdotenv": "~2.4",
         "magento/magento2-functional-testing-framework": "dev-develop"
+        "vlucas/phpdotenv": "~2.4",
```
Comment (dmanners) [inline dev/tests/acceptance/tests/functional/Magento/FunctionalTest/Robots/composer.json]: Seems to be a miss `,` the line above.

File: app/code/Magento/Vault/composer.json [hunk 5]
```diff
@@ -1,15 +1,18 @@
...
         "php": "7.0.2|7.0.4|~7.0.6|~7.1.0",
         "magento/framework": "100.3.*",
-        "magento/module-sales": "100.3.*",
-        "magento/module-store": "100.3.*",
         "magento/module-checkout": "100.3.*",
-        "magento/module-payment": "100.3.*",
         "magento/module-customer": "100.3.*",
+        "magento/module-payment": "100.3.*",
         "magento/module-quote": "100.3.*"
+        "magento/module-sales": "100.3.*",
```
Comment (dmanners) [inline app/code/Magento/Vault/composer.json]: Seems to be a miss `,` the line above.

**Review by dmanners — CHANGES_REQUESTED** [review]
A couple of syntax errors, but other than that it looks good.

File: app/code/Magento/Vault/composer.json (hunk 5 above)
Comment (localheinz) [reply app/code/Magento/Vault/composer.json]: Thank you for catching these, @dmanners!

**Review by dmanners — APPROVED** [review]
Changes look good. Will see what the builds say and let you know.

**Review by dmanners — APPROVED** [review]
Looks good. I am going to run this through our internal systems.

**Comment by localheinz** [comment]
Thank you, @dmanners and @okorshenko!

**Pull request merged**